ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java

```
@@ -0,0 +1,107 @@
+                        "<osm attribution=\"http://www.openstreetmap.org/copyright\" copyright=\"OpenStreetMap and contributors\" generator=\"CGImap 0.6.1 (11633 thorn-03.openstreetmap.org)\" license=\"http://opendatacommons.org/licenses/odbl/1-0/\" version=\"0.6\">" +
+                        "<relation id=\"5848537\"/>" +
+                        "</osm>";
+
+        String resultMinification = osmMinifier.minify(sourceData);
+        minifiedData = normalizationMinifiedData(minifiedData);
+        resultMinification = normalizationResultMinification(resultMinification);
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test(expected = InvalidOSMDataException.class) // minification invalid string osmData
+    public void checkingInvalidStringMinification() throws InvalidOSMDataException {
+
+        final String sourceData =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?> " +
+                        "<osm attribution=\"http://www.openstreetmap.org/copyright\" copyright=\"OpenStreetMap and contributors\" generator=\"CGImap 0.6.1 (11633 thorn-03.openstreetmap.org)\" license=\"http://opendatacommons.org/licenses/odbl/1-0/\" version=\"0.6\">" +
+                        "<way id=\"580429971\" visible=\"true\" version=\"1\" changeset=\"58135270\" timestamp=\"2018-04-16T11:43:32Z\" user=\"Alex_Enot\" uid=\"5757927\"></way>";
+        final String minifiedData = "";
+
+        String resultMinification = osmMinifier.minify(sourceData);
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test(expected = InvalidOSMDataException.class) // minification empty string osmData
+    public void checkingEmptyStringMinification() throws InvalidOSMDataException {
+
+        final String sourceData = "";
+        final String minifiedData = "";
+
+        String resultMinification = osmMinifier.minify(sourceData);
+        Assert.assertEquals(minifiedData, resultMinification);
```

I did not know this design...
'minifiedData' must not be empty, method should throw an exception.